### PR TITLE
Upgrade share upgrade warning notification

### PR DESCRIPTION
**Pull request by nf-s**
### Upgrade share upgrade warning notification

https://github.com/TerriaJS/terriajs/issues/4796

### Testing links

- Short report section ... #clean&https://gist.githubusercontent.com/nf-s/be2c74551c426011914fedd6d3d8b20d/raw/68340b1cc1dff27158402b36dfad836d4f3abaef/shortreportcatalog.json

### Changes

* Make share conversion notification more pretty (moved from `Terria.ts` to `shareConvertNotification.tsx`)
* Tsxify `Collapsible`
* `ShortReportSections` now uses `Collapsible`
* Add `onToggle`, `btnRight`, `btnStyle`, `titleTextProps` and `bodyBoxProps` props in `Collapsible`
* Add `Notification.message` support for `(viewState: ViewState) => React.ReactNode`

### Checklist

-   [ ] There are unit tests to verify my changes are correct or unit tests aren't applicable (if so, write quick reason why unit tests don't exist)
-   [x] I've updated CHANGES.md with what I changed.

### CODE REVIEW

File: lib/Models/Terria.ts [hunk 1]
```diff
@@ -45,25 +50,22 @@ import CommonStrata from "./CommonStrata";
...
 import Workbench from "./Workbench";
-import openGroup from "./openGroup";
-import getDereferencedIfExists from "../Core/getDereferencedIfExists";
-import SplitItemReference from "./SplitItemReference";
-import Internationalization, {
-  I18nStartOptions,
-  LanguageConfiguration
-} from "./Internationalization";
-import { Notification } from "../ReactViewModels/ViewState";
-import CesiumEvent from "terriajs-cesium/Source/Core/Event";
```
Comment (nf-s) [inline lib/Models/Terria.ts:66]: Org imports

File: lib/Language/en/translation.json [hunk 2]
```diff
@@ -170,7 +170,12 @@
     "btnMapShareText": "Share / Print",
     "btnCatalogShareTitle": "Share your catalogue with others",
     "btnStoryShareTitle": "Share your story with others",
-    "btnMapShareTitle": "Share your map with others"
+    "btnMapShareTitle": "Share your map with others",
+    "convertNotificationTitle": "Warning: share link upgraded",
+    "convertNotificationMessage": "This share link is based on an older version of TerriaJS - the software powering this web map. While best efforts are made to preserve the link, you might encounter issues with visualising the data.  \n\nIf you're a user of the share link, click OK to make the warning message disappear and continue with accessing the data.",
+    "convertNotificationHelp": "If you have created the share link, check the Help menu to see tips on updating the share link.",
+    "convertNotificationFeedback": "For questions or feedback, click here.",
+    "convertNotificationWarningsTitle": "Warnings"
```
Comment (soyarsauce) [inline lib/Language/en/translation.json:178]: Thanks for keeping strings in here!

**Review by soyarsauce — APPROVED** [review]

File: lib/ReactViews/Notification/shareConvertNotification.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+      ? messagesForPath[pathString].push(message.message)
+      : (messagesForPath[pathString] = [message.message]);
+  });
+
+  const rootMessages = messagesForPath["root"];
+  delete messagesForPath["root"];
+
+  const showHelp = () => {
+    viewState.showHelpPanel();
+    viewState.selectHelpMenuItem("storymigration");
```
Comment (soyarsauce) [inline lib/ReactViews/Notification/shareConvertNotification.tsx:32]: Also where did you end up getting the key? I was surprised cause from my view it just looks like you guessed it, but of course not

Comment (soyarsauce) [reply lib/ReactViews/Notification/shareConvertNotification.tsx:32]: Oh, I put it in the ticket, of course.

Comment (nf-s) [reply lib/ReactViews/Notification/shareConvertNotification.tsx:32]: stole it from https://github.com/TerriaJS/terriajs/issues/4796#issuecomment-700572601

**Pull request merged**